### ISSUE TITLE
[MNG-6653] DefaultProjectBuildingRequest copy constructor does not copy all fields

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8573]
**[Falko Modler](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=famod)** opened **[MNG-6653](https://issues.apache.org/jira/browse/MNG-6653?redirect=false)** and commented

```java
public DefaultProjectBuildingRequest( ProjectBuildingRequest request )
{
    this();
    setProcessPlugins( request.isProcessPlugins() );
    setProfiles( request.getProfiles() );
    setActiveProfileIds( request.getActiveProfileIds() );
    setInactiveProfileIds( request.getInactiveProfileIds() );
    setSystemProperties( request.getSystemProperties() );
    setUserProperties( request.getUserProperties() );
    setRemoteRepositories( request.getRemoteRepositories() );
    setPluginArtifactRepositories( request.getPluginArtifactRepositories() );
    setRepositorySession( request.getRepositorySession() );
    setLocalRepository( request.getLocalRepository() );
    setBuildStartTime( request.getBuildStartTime() );
    setProject( request.getProject() );
    setResolveDependencies( request.isResolveDependencies() );
    setValidationLevel( request.getValidationLevel() );
}
```

This is missing `setRepositoryMerging( request.getRepositoryMerging() );` and `setResolveVersionRanges( request.isResolveVersionRanges() );`.

---

**Affects:** 3.6.1

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The copy constructor isn't used in the Core codebase, but for the sake of completeness, this should be fixed.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [dd0776d949bf91abaeeb0af0c150288d52cd60f4](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=dd0776d949bf91abaeeb0af0c150288d52cd60f4).
